client/client.go

```diff
@@ -32,6 +33,29 @@ import (
 	"github.com/canonical/pebble/internal/wsutil"
 )
 
+// SocketNotFoundError is the error type returned when the Pebble client fails
+// to find a unix socket at the specified path.
+type SocketNotFoundError struct {
+	// Error wrapped error.
+	Err error
+
+	// SocketPath of the non existent socket.
+	SocketPath string
```

Comment:
Naming: the type already includes the word "socket", so this could be called just `Path`.